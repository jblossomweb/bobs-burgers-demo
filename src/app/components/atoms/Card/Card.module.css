.container {
}

.card {
  @apply min-w-0 rounded-lg ring-1 ring-black ring-opacity-5 overflow-hidden;
  @apply bg-white flex flex-col justify-between;
}

.body {
  @apply p-4 h-2/3;
}

.image {
  @apply object-cover w-1/3;
}

.title {
  @apply mb-4 font-semibold text-gray-800 text-xl;
}

.content {
  @apply text-gray-600 text-sm;
}
