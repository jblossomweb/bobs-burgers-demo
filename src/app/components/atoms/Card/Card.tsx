import React from 'react'
import clsx from 'clsx'
import style from './Card.module.css'

export interface Props {
  title?: string
  children?: React.ReactNode
  head?: React.ReactNode
  foot?: React.ReactNode
  width?: number
  className?: React.HTMLAttributes<HTMLElement>['className']
}

const widthStyle = (width: Props['width']): React.CSSProperties =>
  width ? { width: `${width}px` } : {}

const Card: React.FC<Props> = ({
  title,
  children,
  width,
  head,
  foot,
  className = '',
}) => (
  <div className={style.container} style={widthStyle(width)}>
    <div className={clsx(style.card, className)}>
      {head}
      {!!(title || children) && (
        <div className={style.body}>
          {!!title && <h2 className={style.title}>{title}</h2>}
          {!!children && <div className={style.content}>{children}</div>}
        </div>
      )}
      {foot}
    </div>
  </div>
)

export default Card
