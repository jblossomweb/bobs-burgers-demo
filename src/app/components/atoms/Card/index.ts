/* eslint-disable no-restricted-exports */
export type { Props } from './Card'
export { default } from './Card'
