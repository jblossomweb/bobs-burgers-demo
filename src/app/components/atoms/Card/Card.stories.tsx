import React from 'react'
import type { ComponentStory, ComponentMeta } from '@storybook/react'
import Button from 'app/components/atoms/Button'

import Card, { type Props } from './Card'

export default {
  title: 'components/atoms/Card',
  component: Card,
  argTypes: {
    title: { control: 'text' },
    width: { control: 'number' },
    children: { control: 'text' },
    head: { control: 'none' },
    foot: { control: 'none' },
  },
} as ComponentMeta<typeof Card>

const Template: ComponentStory<typeof Card> = ({
  title,
  width,
  head,
  foot,
  children,
}: Props) => (
  <Card title={title} width={width} head={head} foot={foot}>
    {children}
  </Card>
)

// storybook borks on the next image when injected through a prop
// so we use a regular img tag
const renderImage = (width?: number) => (
  <img src="https://picsum.photos/600/400" width={width} alt="sample" />
)

const renderButton = () => <Button>Button</Button>

export const Sample = Template.bind({})
Sample.args = {
  title: 'Sample Card',
  children:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
}

export const Sized = Template.bind({})
Sized.args = {
  ...Sample.args,
  width: 400,
}

export const WithImage = Template.bind({})
WithImage.args = {
  ...Sample.args,
  width: 400,
  head: renderImage(400),
}

export const WithButton = Template.bind({})
WithButton.args = {
  ...Sample.args,
  width: 400,
  foot: renderButton(),
}

export const WithBoth = Template.bind({})
WithBoth.args = {
  ...Sample.args,
  width: 400,
  head: renderImage(400),
  foot: renderButton(),
}
