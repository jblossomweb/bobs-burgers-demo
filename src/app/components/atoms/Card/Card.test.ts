import storySnaps from 'test/utils/storySnaps'
import * as stories from './Card.stories'

describe('app/components/atoms/Card', () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
