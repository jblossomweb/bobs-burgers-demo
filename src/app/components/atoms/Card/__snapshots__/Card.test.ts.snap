// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/atoms/Card stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="card"
    >
      <div
        class="body"
      >
        <h2
          class="title"
        >
          Sample Card
        </h2>
        <div
          class="content"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/components/atoms/Card stories Sized matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    style="width: 400px;"
  >
    <div
      class="card"
    >
      <div
        class="body"
      >
        <h2
          class="title"
        >
          Sample Card
        </h2>
        <div
          class="content"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/components/atoms/Card stories WithBoth matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    style="width: 400px;"
  >
    <div
      class="card"
    >
      <img
        alt="sample"
        src="https://picsum.photos/600/400"
        width="400"
      />
      <div
        class="body"
      >
        <h2
          class="title"
        >
          Sample Card
        </h2>
        <div
          class="content"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
      </div>
      <button
        class="button"
        type="button"
      >
        Button
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/components/atoms/Card stories WithButton matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    style="width: 400px;"
  >
    <div
      class="card"
    >
      <div
        class="body"
      >
        <h2
          class="title"
        >
          Sample Card
        </h2>
        <div
          class="content"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
      </div>
      <button
        class="button"
        type="button"
      >
        Button
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/components/atoms/Card stories WithImage matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    style="width: 400px;"
  >
    <div
      class="card"
    >
      <img
        alt="sample"
        src="https://picsum.photos/600/400"
        width="400"
      />
      <div
        class="body"
      >
        <h2
          class="title"
        >
          Sample Card
        </h2>
        <div
          class="content"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
